docs/DynamicInputTextures.md

```diff
@@ -0,0 +1,139 @@
+# Dolphin Dynamic Input Textures Specification (v1)
+
+## Format
+Dynamic Input Textures are generated textures based on a user's input formed from a group of png files and json files.
+
+```
+\__ Dolphin User Directory
+    \__ Load (Directory)
+        \__ DynamicInputTextures (Directory)
+            \__ GAMEID (Directory)
+                \__ PNG and JSON GO HERE
+```
+
+
+``GAMEID`` can be one or multiple directories which are named after:
+* a complete Game ID (e.g. ``SMNE01`` for "New Super Mario Bros. Wii (NTSC)")
+* one without a region (e.g. ``SMN`` for "New Super Mario Bros. Wii (All regions)").
+
+### PNG files
+
+At a minimum two images are required to support the generation and any number of 'button' images.  These need to be in PNG format.
+
+### JSON files
+
+You need at least a single json file that describes the generation parameters.  You may have multiple JSON files if you prefer that from an organizational standpoint.
+
+#### Possible fields in the JSON for a texture
+
+In each json, one or more generated textures can be specified.  Each of those textures can have the following fields:
+
+|Identifier               |Required | Since |
+|-------------------------|---------|-------|
+|``colorkey_image``       | **Yes** | v1    |
+|``fallback_image``       | **Yes** | v1    |
+|``emulated_controls``    | **Yes** | v1    |
+|``device``               | No      | v1    |
+|``host_controls``        | No      | v1    |
+
+*colorkey_image* - an image with colored boxes, will be used as a base image, so any non-input related sections can be upscaled/redrawn if desired.
+
+*fallback_image* - after looking at the ``colorkey_image``, if the color isn't mapped to a specific button, it will use this image as a reference for what to put in the color. This should be the same size/quality as the ``colorkey_image`` but without the colored boxes.
+
+*emulated_controls* - a series of emulated buttons that map to a given color rectangle.
+
```

Comment:
Did you intentionally leave _two_ blank lines here between explainations, vs. one between every other?

Reply:
Whoops, no that wasn't intentional